samtranslator/translator/arn_generator.py

```diff
@@ -43,7 +49,17 @@ def get_partition_name(cls, region=None):
         if region is None:
```

Comment:
Shall we move to `if not X` tests in here to also catch empty strings?

Reply:
My goal was not to adjust the current behavior in this pr but instead focus on providing a better interface for tools (like SAM CLI), who get a profile and region through a command line. There is probably a case to validate the region in this manner but we aren't changing that expectation here at all. I don't think boto3 has a way to do this, so we would have to keep a list of regions. For now, I would prefer to keep the behavior in line with the previous (default to aws partition on non-region strings) and later add in this further validation if needed.